src/TreeSelect.tsx

```diff
@@ -120,5 +119,7 @@ export default class TreeSelect extends React.Component<Props, State> {
   private reportChanged(items: Items) {
     this.props.onChange(findAll(items, i => i.selected && i.selectable)
       .map(i => i.original) as PublicItem[]);
+    this.setState({ filter: null });
+    if (this.filterInput) this.filterInput.focus();
```

Comment:
I am not familiar with the codebase, so the only thing I can say here is that maybe you could call the filterInput focus on the callback from the setState method.
You will probably know if that makes sense or not.

Reply:
@AquiGorka is right to question this. This is an imperative logic that tries to catch when to focus the field. But putting it into the `setState` callback does not change the fundamental. Instead, it’d be better to modify the `componentDidUpdate` logic instead. @chadoh, could you see if you can express your intent [in there](https://github.com/citrusbyte/tree-select/blob/28cbd6f0f0ae034d811fdfd310e1882e368e0973/src/TreeSelect.tsx#L60)?

Reply:
Ok, I altered `componentDidUpdate` to stop checking the previous props. I'm not sure why that check was there; everything I've thought to check still works well. Let me know if you can think of something else that this may have broken, @denis-sokolov.